smarts/core/scenario.py

```diff
@@ -364,6 +362,27 @@ def variations_for_all_scenario_roots(
                     traffic_history=concrete_traffic_history,
                 )
 
+    @staticmethod
+    def discover_traffic_history_scenarios_count(
+        scenarios_or_scenarios_dirs: Sequence[str],
+    ):
+        scenario_roots = []
+        for root in scenarios_or_scenarios_dirs:
+            if Scenario.is_valid_scenario(root):
+                # This is the single scenario mode, only training against a single scenario
+                scenario_roots.append(root)
+            else:
+                scenario_roots.extend(Scenario.discover_scenarios(root))
+
+        total_traffic_history_variations = 0
+        for scenario_root in scenario_roots:
+            traffic_histories = Scenario.discover_traffic_histories(scenario_root) or [
+                None
+            ]
+            if traffic_histories != [None]:
+                total_traffic_history_variations += len(traffic_histories)
+        return total_traffic_history_variations
```

Comment:
My problem with this method is that the information comes at non-trivial cost and then that cost must be incurred again in order to make use of the scenarios.

The exact number of scenarios is useful but the locations of those scenarios are also tied in for the majority of uses.